docs/en/02_Developer_Guides/15_Customising_the_Admin_Interface/07_ReactJS_Redux_and_GraphQL.md

```diff
@@ -0,0 +1,666 @@
+
+__my-module/js/main.js__
+
+```js
+Injector.transform(
+	'my-transformation',
+	(updater) => {
+		updater.react('MyComponent', MyCustomComponent);
+		updater.reducer('myCustom', MyCustomReducer);
+
+	},
+	{ after: 'another-module' }
+);
+
+```
+
+`before` and `after` also accept arrays of constraints.
+
+```js
+Injector.transform(
+  'my-transformation', 
+  (updater) => updater.react('MyComponent', MyCustomComponent);
+  { before: ['my-transformation', 'some-other-transformation'] }
+);
+```
+
+### Using the * flag
+
+If you really want to be sure your customisation gets loaded first or last, you can use 
+`*` as your `before` or `after` reference. 
+
+```js
+Injector.transform(
+  'my-transformation', 
+  (updater) => updater.react('MyComponent', FinalTransform),
+  { after: '*' }
+);
+```
+<div class="info" markdown="1">
+This flag can only be used once per transformation.
+The following are not allowed:
+* `{ before: ['*', 'something-else'] }`
+* `{ after: '*', before: 'something-else' }`
+</div>
+
+## Injector context
+
+Because so much of UI design depends on context, dependency injection in the frontend is not necessarily universal. Instead, services are fetched with context.
+
+_example_:
+```js
+const CalendarComponent = Injector.get('Calendar', 'AssetAdmin.FileEditForm.StartDate');
+```
+
+Likewise, services can be applied for specific contexts.
+
+```js
+Injector.transform('my-transform', (updater) => {
+	// Applies to all text fields in AssetAdmin
+	updater.react('TextField.AssetAdmin', MyTextField);
+
+	// Applies to all text fields in AssetAdmin editform
+	updater.react('TextField.AssetAdmin.FileEditForm', MyTextField);
+
+	// Applies to any textfield named "Title" in AssetAdmin
+	updater.react('TextField.AssetAdmin.*.Title', MyTextField);
+
+	// Applies to any textfield named "Title" in any admin
+	updater.react('TextField.*.*.Title', MyTextField);
+
+	// Applies to all text fields in AssetAdmin
+	updater.react('TextField.AssetAdmin', MyTextField);
+
+})
+```
+
+ To apply context-based transformations, you'll need to know the context of the component you want to customise. To learn this,
+ open your React Developer Tools (see above) window and inspect the component name. The
+ context of the component is displayed between two square brackets, appended to the original name, for example:
+ `TextField[TextField.AssetAdmin.FileEditForm.Title]`. The context description is hierarchical, starting
+ with the most general category (in this case, "Admin") and working its way down to the most specific
+ category (Name = 'Title'). You can use Injector to hook into the level of specificity that you want.
+
+
+# Customising React components with Injector
+
+When middleware is used to customise a React component, it is known as a [higher order component](https://facebook.github.io/react/docs/higher-order-components.html).
```

Comment:
You mention HoCs multiple times before this - put the link and explanation in earlier?